dbms/src/Functions/FunctionsHashing.h

```diff
@@ -956,6 +956,136 @@ class FunctionURLHash : public IFunction
     }
 };
 
+struct JavaHashImpl
+{
+    static Int32 apply(const char * data, const size_t size)
+    {
+        Int32 h = 0;
+        for (int i = 0; i < (int)size; ++i)
+        {
+            h = 31 * h + data[i];
+        }
+        return h;
+    }
+};
+
+/*
+ * the java string hash implement,
+ * many system from java world use this string hash function or based it
+ */
+class FunctionJavaHash : public IFunction
```

Comment:
Why don't use FunctionIntHash template?

Reply:
FunctionIntHash is a hash form 'int -> int', but JavaHash is 'string -> int'

Reply:
Why don't use FunctionAnyHash template?